legacy/firmware/transaction.c

```diff
@@ -288,6 +288,12 @@ static int address_to_script_pubkey(const CoinInfo *coin, const char *address,
     // push addr_raw (segwit_addr_decode makes sure addr_raw_len is at most 40)
     script_pubkey[0] = witver == 0 ? 0 : 80 + witver;
     script_pubkey[1] = addr_raw_len;
+    // check that P2TR address encodes a valid BIP340 public key
+    if (witver == 1 && addr_raw_len == 32) {
```

Comment:
I am wondering about the purpose of checking `addr_raw_len == 32` at this point. What happens if the length is not 32 bytes? The public key check won't be performed and the address will pass validation, which is not what we want.

Reply:
If we want to prevent the user from burning funds by sending them to an unrecognized address, then:
- We should be checking the witness program length somewhere. Either in `zkp_bip340_verify_publickey()`, i.e. remove the `addr_raw_len == 32` check above, or in `segwit_addr_decode()`, as we do for v0 witness programs.
- We should also be checking that we recognize the witness version 0 or 1, which we don't seem to be doing in T1

Reply:
https://github.com/trezor/trezor-firmware/pull/2077/commits/ee590674cae5284489ab78aefa7ab2906d564252